Tupan tests fail with numpy >=2.4.0

**Describe the bug**
The tupan tests fail with numpy >= 2.4.0

**To Reproduce**
`./setup test tupan`

**Expected behavior**
Tests passed

**Logs**
```
=================================== FAILURES ===================================
__________________________ TestTupanInterface.test02 ___________________________

self = <amuse.rfi.core.CodeFunction object at 0x7240eef8cf50>
arguments_list = (), keyword_arguments = {}, dtype_to_values = {}
handle_as_array = False, call_id = 784

    def __call__(self, *arguments_list, **keyword_arguments):
        if self.interface.async_request:
            try:
                self.interface.async_request.wait()
            except Exception as ex:
                warnings.warn("Ignored exception in async call: " + str(ex))
    
        dtype_to_values = self.converted_keyword_and_list_arguments(
            arguments_list, keyword_arguments
        )
    
        handle_as_array = self.must_handle_as_array(dtype_to_values)
    
        if not self.owner is None:
            CODE_LOG.info(
                "start call '%s.%s'", self.owner.__name__, self.specification.name
            )
    
        call_id = random.randint(0, 1000)
    
        try:
            self.interface.channel.send_message(
                call_id, self.specification.id, dtype_to_arguments=dtype_to_values
            )
    
>           dtype_to_result = self.interface.channel.recv_message(
                call_id, self.specification.id, handle_as_array
            )

/home/lourens/.miniconda3/envs/Amuse-2026.3.0/lib/python3.14/site-packages/amuse/rfi/core.py:129: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <amuse.rfi.channel.MpiChannel object at 0x724057d11310>, call_id = 784
function_id = 20920053, handle_as_array = False, has_units = False

    def recv_message(self, call_id, function_id, handle_as_array, has_units=False):
    
        if self._communicated_splitted_message:
            x = self._merged_results_splitted_message
            self._communicated_splitted_message = False
            del self._merged_results_splitted_message
            return x
    
        message = ServerSideMPIMessage(
            polling_interval=self.polling_interval_in_milliseconds * 1000
        )
        try:
            message.receive(self.intercomm)
        except MPI.Exception as ex:
            self._is_inuse = False
            self.stop()
            raise ex
        self.inuse_semaphore.acquire()
        try:
            if not self._is_inuse:
                raise exceptions.CodeException(
                    "You've tried to recv a message to a code that is not handling a message, this is not correct"
                )
            self._is_inuse = False
        finally:
            self.inuse_semaphore.release()
    
        if message.error:
            error_message = (
                message.strings[0] if len(message.strings) > 0 else "no error message"
            )
            if message.call_id != call_id or message.function_id != function_id:
                self.stop()
                error_message += " - code probably died, sorry."
>           raise exceptions.CodeException("Error in code: " + error_message)
E           amuse.support.exceptions.CodeException: Error in code: setting an array element with a sequence.

/home/lourens/.miniconda3/envs/Amuse-2026.3.0/lib/python3.14/site-packages/amuse/rfi/channel.py:1361: CodeException

During handling of the above exception, another exception occurred:

self = <test_tupan.TestTupanInterface testMethod=test02>

    def test02(self):
        if MODULES_MISSING:
            self.skip("Failed to import a module required for Tupan")
        print("Test TupanInterface new_particle / get_state")
        instance = self.new_instance_of_an_optional_code(TupanInterface)
        self.assertEqual(0, instance.initialize_code())
        self.assertEqual(0, instance.commit_parameters())
    
        id, error = instance.new_particle(mass=11.0, radius=2.0, x=0.0, y=0.0, z=0.0, vx=0.0, vy=0.0, vz=0.0)
        self.assertEqual(0, error)
        self.assertEqual(0, id)
        id, error = instance.new_particle(mass=21.0, radius=5.0, x=10.0, y=0.0, z=0.0, vx=10.0, vy=0.0, vz=0.0)
        self.assertEqual(0, error)
        self.assertEqual(1, id)
>       self.assertEqual(0, instance.commit_particles())
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^

amuse_tupan/tests/test_tupan.py:41: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <amuse.rfi.core.CodeFunction object at 0x7240eef8cf50>
arguments_list = (), keyword_arguments = {}, dtype_to_values = {}
handle_as_array = False, call_id = 784

    def __call__(self, *arguments_list, **keyword_arguments):
        if self.interface.async_request:
            try:
                self.interface.async_request.wait()
            except Exception as ex:
                warnings.warn("Ignored exception in async call: " + str(ex))
    
        dtype_to_values = self.converted_keyword_and_list_arguments(
            arguments_list, keyword_arguments
        )
    
        handle_as_array = self.must_handle_as_array(dtype_to_values)
    
        if not self.owner is None:
            CODE_LOG.info(
                "start call '%s.%s'", self.owner.__name__, self.specification.name
            )
    
        call_id = random.randint(0, 1000)
    
        try:
            self.interface.channel.send_message(
                call_id, self.specification.id, dtype_to_arguments=dtype_to_values
            )
    
            dtype_to_result = self.interface.channel.recv_message(
                call_id, self.specification.id, handle_as_array
            )
        except Exception as ex:
            CODE_LOG.info(
                "Exception when calling function '{0}', of code '{1}', exception was '{2}'".format(
                    self.specification.name, type(self.interface).__name__, ex
                )
            )
>           raise exceptions.CodeException(
                "Exception when calling function '{0}', of code '{1}', exception was '{2}'".format(
                    self.specification.name, type(self.interface).__name__, ex
                )
            )
E           amuse.support.exceptions.CodeException: Exception when calling function 'commit_particles', of code 'TupanInterface', exception was 'Error in code: setting an array element with a sequence.'

/home/lourens/.miniconda3/envs/Amuse-2026.3.0/lib/python3.14/site-packages/amuse/rfi/core.py:138: CodeException
----------------------------- Captured stdout call -----------------------------
Test TupanInterface new_particle / get_state
```

**Environment (please complete the following information):**
 - OS and version: Ubuntu 24.04
 - Environment: Conda

**Additional context**
`conda install 'numpy<2.4'` makes things work again, but is obviously not a long-term solution.

--- a/src/amuse_tupan/interface.py
+++ b/src/amuse_tupan/interface.py
@@ -572,17 +572,17 @@ class Tupan(GravitationalDynamics, GravityFieldCode):
     def define_methods(self, handler):
         GravitationalDynamics.define_methods(self, handler)
 
-        handler.add_method(
-            "get_eta",
-            (),
-            (handler.NO_UNIT, handler.ERROR_CODE,)
-        )
+        # handler.add_method(
+        #     "get_eta",
+        #     (),
+        #     (handler.NO_UNIT, handler.ERROR_CODE,)
+        # )
 
-        handler.add_method(
-            "set_eta",
-            (handler.NO_UNIT,),
-            (handler.ERROR_CODE,)
-        )
+        # handler.add_method(
+        #     "set_eta",
+        #     (handler.NO_UNIT,),
+        #     (handler.ERROR_CODE,)
+        # )
 
         handler.add_method(
             "get_time",

--- a/src/amuse_tupan/interface.py
+++ b/src/amuse_tupan/interface.py
@@ -572,18 +572,6 @@ class Tupan(GravitationalDynamics, GravityFieldCode):
     def define_methods(self, handler):
         GravitationalDynamics.define_methods(self, handler)
 
-        # handler.add_method(
-        #     "get_eta",
-        #     (),
-        #     (handler.NO_UNIT, handler.ERROR_CODE,)
-        # )
-
-        # handler.add_method(
-        #     "set_eta",
-        #     (handler.NO_UNIT,),
-        #     (handler.ERROR_CODE,)
-        # )
-
         handler.add_method(
             "get_time",
             (),

--- a/src/amuse_tupan/interface.py
+++ b/src/amuse_tupan/interface.py
@@ -1,10 +1,10 @@
 
-from amuse.datamodel.particles import Particle, Particles
 from amuse.community import *
 from amuse.community.interface.gd import GravitationalDynamicsInterface
 from amuse.community.interface.gd import GravitationalDynamics
 from amuse.community.interface.gd import SinglePointGravityFieldInterface
 from amuse.community.interface.gd import GravityFieldCode
+from amuse.datamodel.particles import Particle, Particles
 from amuse.rfi.core import PythonCodeInterface
 
 import sys
@@ -62,7 +62,7 @@ class TupanImplementation(object):
     def commit_particles(self):
         ps = ParticleSystem(nstars=len(self.particles))
         for (i, p) in enumerate(self.particles):
-            ps.id[i] = i
+            ps.id[i] = p.id
             ps.mass[i] = p.mass
             ps.radius[i] = p.radius   # XXX: 'radius' is not yet used in Tupan.
             ps.eps2[i] = self.eps2/2
@@ -95,6 +95,7 @@ class TupanImplementation(object):
         vx, vy, vz,
     ):
         ps = Particle()
+        ps.id = len(self.particles)
         ps.mass = mass
         ps.radius = radius
         ps.x = x
@@ -103,7 +104,7 @@ class TupanImplementation(object):
         ps.vx = vx
         ps.vy = vy
         ps.vz = vz
-        index_of_the_particle.value = len(self.particles)
+        index_of_the_particle.value = ps.id
         self.particles.add_particle(ps)
 
         return 0

--- a/src/amuse_tupan/interface.py
+++ b/src/amuse_tupan/interface.py
@@ -1,5 +1,5 @@
 
-
+from amuse.datamodel.particles import Particle, Particles
 from amuse.community import *
 from amuse.community.interface.gd import GravitationalDynamicsInterface
 from amuse.community.interface.gd import GravitationalDynamics
@@ -38,7 +38,7 @@ class TupanImplementation(object):
         self.integrator_method = "sia21h.dkd"
         self.pn_order = 0
         self.clight = None
-        self.particles = []
+        self.particles = Particles()
         self.particles_initialized = False
 
     def initialize_code(self):
@@ -60,43 +60,52 @@ class TupanImplementation(object):
         return 0
 
     def commit_particles(self):
-        num = len(self.particles)
-        ps = ParticleSystem(nstars=num)
+        ps = ParticleSystem(nstars=len(self.particles))
         for (i, p) in enumerate(self.particles):
             ps.id[i] = i
             ps.mass[i] = p.mass
             ps.radius[i] = p.radius   # XXX: 'radius' is not yet used in Tupan.
             ps.eps2[i] = self.eps2/2
-            ps.rx[i] = p.rx
-            ps.ry[i] = p.ry
-            ps.rz[i] = p.rz
+            ps.rx[i] = p.x
+            ps.ry[i] = p.y
+            ps.rz[i] = p.z
             ps.vx[i] = p.vx
             ps.vy[i] = p.vy
             ps.vz[i] = p.vz
-        self.integrator = Integrator(self.eta,
-                                     self.time_begin,
-                                     ps,
-                                     method=self.integrator_method,
-                                     pn_order=self.pn_order,
-                                     clight=self.clight)
+
+        self.integrator = Integrator(
+            self.eta,
+            self.time_begin,
+            ps,
+            method=self.integrator_method,
+            pn_order=self.pn_order,
+            clight=self.clight
+        )
         return 0
 
     def synchronize_model(self):
         return 0
 
-    def new_particle(self, index_of_the_particle,
-                     mass, radius, x, y, z, vx, vy, vz):
-        ps = ParticleSystem(nstars=1)
-        ps.mass[0] = mass
-        ps.radius[0] = radius
-        ps.rx[0] = x
-        ps.ry[0] = y
-        ps.rz[0] = z
-        ps.vx[0] = vx
-        ps.vy[0] = vy
-        ps.vz[0] = vz
-        self.particles.append(ps)
-        index_of_the_particle.value = len(self.particles)-1
+    def new_particle(
+        self,
+        index_of_the_particle,
+        mass,
+        radius,
+        x, y, z,
+        vx, vy, vz,
+    ):
+        ps = Particle()
+        ps.mass = mass
+        ps.radius = radius
+        ps.x = x
+        ps.y = y
+        ps.z = z
+        ps.vx = vx
+        ps.vy = vy
+        ps.vz = vz
+        index_of_the_particle.value = len(self.particles)
+        self.particles.add_particle(ps)
+
         return 0
 
     def set_state(self, index_of_the_particle,

--- a/src/amuse_tupan/packages/amuse-tupan/pyproject.toml
+++ b/src/amuse_tupan/packages/amuse-tupan/pyproject.toml
@@ -3,8 +3,7 @@ name = "amuse-tupan"
 dynamic = ["version"]
 requires-python = ">=3.7"
 dependencies = [
-    "amuse-framework",
-    "amuse_tupan"
+    "amuse-framework"
 ]
 
 [build-system]
@@ -44,4 +43,3 @@ pythonpath = ["amuse_tupan/tests/"]
 testpaths = ["amuse_tupan/tests"]
 
 addopts = "--import-mode=append"    # test the installed package
-

--- a/src/amuse_tupan/tests/test_tupan.py
+++ b/src/amuse_tupan/tests/test_tupan.py
@@ -174,52 +174,24 @@ class TestTupan(TestWithMPI):
         instance.cleanup_code()
         instance.stop()
 
-    def xtest02(self):
+    def test02(self):
         if MODULES_MISSING:
             self.skip("Failed to import a module required for Tupan")
-        print("Testing Tupan parameters")
+        print("Testing Tupan default parameters")
         instance = self.new_instance_of_an_optional_code(Tupan, self.default_converter)
         instance.initialize_code()
 
-        self.assertEqual(instance.parameters.epsilon_squared,
-            instance.unit_converter.to_si(0.0 | nbody_system.length**2))
-        self.assertEqual(instance.parameters.timestep_parameter, 0.125)
-
-        for par, value in [('epsilon_squared_star_star', 0.0 | nbody_system.length**2),
-                ('epsilon_squared_star_blackhole', 0.0 | nbody_system.length**2),
-                ('epsilon_squared_blackhole_blackhole', 0.0 | nbody_system.length**2),
-                ('initial_timestep_parameter', 1.0e-4),
-                ('timestep_parameter_stars', 0.1),
-                ('timestep_parameter_supermassive_black_holes', 0.4),
-                ('timestep_parameter_intermediate_mass_black_holes', 0.4),
-                ('max_relative_energy_error', 5.0e-5),
-                ('maximum_timestep', 1.0/1024.0 | nbody_system.time),
-                ('smbh_mass', 1.0 | nbody_system.mass)]:
-            self.assertEqual(instance.unit_converter.to_si(value),
-                getattr(instance.parameters, par))
-            setattr(instance.parameters, par, 3.0 | value.unit)
-            self.assertEqual(instance.unit_converter.to_si(3.0 | value.unit),
-                getattr(instance.parameters, par))
-
-        # epsilon_squared is an alias for epsilon_squared_star_star, so epsilon_squared also has become 3:
-        self.assertEqual(instance.parameters.epsilon_squared,
-            instance.unit_converter.to_si(3.0 | nbody_system.length**2))
-        instance.parameters.epsilon_squared = 0.1 | nbody_system.length**2
-        self.assertEqual(instance.parameters.epsilon_squared,
-            instance.unit_converter.to_si(0.1 | nbody_system.length**2))
-        # timestep_parameter is an alias for timestep_parameter_stars, so timestep_parameter also has become 3:
-        self.assertEqual(instance.parameters.timestep_parameter, 3.0)
-        instance.parameters.timestep_parameter = 0.01
-        self.assertEqual(instance.parameters.timestep_parameter, 0.01)
-
-        self.assertEqual(instance.parameters.include_smbh, False)
-        instance.parameters.include_smbh = True
-        self.assertEqual(instance.parameters.include_smbh, True)
-        self.assertEqual(instance.parameters.calculate_postnewtonian, True)
-        instance.parameters.calculate_postnewtonian = False
-        self.assertEqual(instance.parameters.calculate_postnewtonian, False)
-
-        self.assertEqual(instance.parameters.drink, "Vodka martini. Shaken, not stirred.")
+        self.assertEquals(instance.parameters.timestep_parameter, 0.03125)
+        self.assertEquals(
+            instance.parameters.epsilon_squared,
+            instance.unit_converter.to_si(0.0 | nbody_system.length**2)
+        )
+        self.assertEquals(
+            instance.parameters.begin_time,
+            instance.unit_converter.to_si(0.0 | nbody_system.time)
+        )
+        self.assertEquals(instance.parameters.integrator_method, 'sia21h.dkd')
+        self.assertEquals(instance.parameters.pn_order, 0)
 
         instance.stop()
 

